### PR TITLE
chore(deps): upgrade dependencies

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

This PR upgrades almost all upgradeable dependencies. It **does not** upgrade the following dependencies due to issues:

- jest-cli (tests run much much slower and some fail with timeouts for what ever reason after upgrade to v27)
- node-fetch (instead of switching to ESM, we'll wait a little for fetch to land in Node)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -172,8 +172,8 @@
     "supports es6-module and supports css-variables and last 2 versions"
   ],
   "optionalDependencies": {
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0"
```
Comment (renet) [inline package.json:176]: I'm not sure what this actually means, but I think we should support at least react 17+, maybe even more. Is this still possible with this change?

**Review by renet — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (borisdiakur) [reply package.json:176]: Yes, I tested this using liquid-sandbox-react-tailwind linking liquid via npm. No issues.

Comment (renet) [reply package.json:176]: Feel free to merge as soon as you updated the dependencies as talked about in the daily.

**Pull request merged**